### DISCUSSION TITLE
How do you collect multiple rows into a struct?

### DISCUSSION THREAD

**Discussion by DanielCoder834** [jackc/pgx#2096] [Q&A] [answered]
I am trying to do a select query on two columns with a left join clause, and then returning the two columns values. 

This is the code I have so far for the query: 
```
gettingNotesQuery := `SELECT notes.title, notes.notes_id FROM notes
    LEFT JOIN "user" ON "user"."user_id" = notes."user_id" AND notes."user_id" = @UserId;`
	args := pgx.NamedArgs{
		"UserId": userId,
	}
	noteRows, queryErr := dbpool.Db.Query(ctx, gettingNotesQuery, args)
	if queryErr != nil {
		return nil, queryErr
	}
	notes, err := pgx.CollectRows(noteRows, pgx.RowTo[types.Notes])
	if err != nil {
		return nil, err
	}

    return notes, nil
```

This is the code that represents the note being collected: 
```
type Notes struct {
	NoteId string
	Title  string
}
```

I got this error: `error: number of field descriptions must equal number of destinations, got 2 and 1`
It seems this [post](https://github.com/jackc/pgx/issues/985) talks about the issue but it uses the scan call and not the collect rows call. Any help would appreciated. Thanks

**Comment by jackc** [comment, marked as answer]
You probably want `pgx.RowToStructByPos` instead of `pgx.RowTo`.

**Reply by DanielCoder834** [reply]
Thank you. It worked
